doc/specs/#2325 - Default Profile Settings.md

```diff
@@ -0,0 +1,314 @@
+            "acrylicOpacity": 0.1,
+            "fontFace": "Cascadia Code",
+            "fontSize": 10
+        },
+        {
+            "guid": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
+            "inheritFrom": "{11111111-1111-1111-1111-111111111111}",
+            "name": "Windows PowerShell",
+            "commandline": "powershell.exe",
+            "hidden": false
+        },
+        {
+            "guid": "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}",
+            "inheritFrom": "{11111111-1111-1111-1111-111111111111}",
+            "name": "cmd",
+            "commandline": "cmd.exe",
+            "hidden": false
+        },
+        {
+            "guid": "{0caa0dad-ffff-5f56-a8ff-afceeeaa6101}",
+            "inheritFrom": "{0caa0dad-35be-5f56-a8ff-afceeeaa6101}",
+            "name": "This is another CMD",
+            "commandline": "cmd.exe /c myCoolScript.bat",
+            "hidden": false
+        }
+    ],
+    "schemes": [],
+    "keybindings": []
+}
+```
+
+#### Benefits
+
+##### Matches the existing settings model without major refactoring
+Simply adding a new property to `Profile` would not majorly alter the structure
+of the file.
+
+##### Property name is unique
+`inheritFrom` is very unique relative to other keys we already have.
+
+##### Powerful
+This lets the user have potentially many layers of settings grouping. hese
+layers would let the user seperate out common settings however they like,
+without forcing them to a single "default" profile. They could potentially have
+many "default" profiles, e.g.
+* one that's used for all their WSL profiles, with `startingDirectory` set to
+  `~` and `fontFace` set to "Ubuntu Mono"
+* One that's used for all their powershell profiles
+
+etc.
+
+#### Concerns
+
+##### GUIDs are not human friendly
+
+Using the guid in the `inheritFrom` field is the only way to be sure we're
+uniquely identifying profiles. However, guids are notoriously un-friendly. The
+above example manually uses `"{11111111-1111-1111-1111-111111111111}"` as the
+guid of the "default" profile, but inheriting from other profiles with "real"
+GUIDs would be less understandable. Consider the "This is another CMD" case,
+where it's inheriting from the "cmd" profile. That `"inheritFrom"` value does
+not mean at a quick glance "cmd".
+
+##### We have to make sure that there are no cycles as we're layering
+
+This is mostly a technical challenge, but this does make the implementation a
+bit trickier.
+
+##### How does this work with the settings UI?
+
+When the user edits settings for a profile with the UI, do we only place the
+changes in the top-most profile?
+
+How do we communicate in the UI that a profile is inheriting settings from other
+profiles?
+
+##### Harder to mentally parse
+Maybe not as easy to mentally picture how one profile inherits from another. The
+user would probably need to manually build the tree of profile inheritance in
+their own head to understand how a profile gets its settings.

```

Comment:
One additional concern:

I still want a place to define a _default settings_ block that isn't one of my profiles.

* If it has to be a real live profile, would it be.. hidden=true?
   * Would the inheritee inherit the hidden=true?
   * If it wasn't, what would stop it from showing up in my list?
* If it had to be a real profile, would I then have to override the commandline?

Imagine this scenario:

```jsonc
"profiles": [
	{
		"guid": "{abcd}",
		"name": "cmd",
		"fontFace": "Cascadia Code",
		"background": "#f00ff0",
		"commandline": "cmd.exe"
	},
	{
		"guid": "{defg}",
		"name": "WSL",
		"inheritsFrom": "{abcd}", // inherit from cmd because i put my defaults in there
		"source": "Microsoft.Terminal.Wsl" // AUTOGENERATED DYNAMIC PROFILE
	}
]
```

Does "WSL" inherit the `commandline`? It should, since it inherits everything from cmd.


**we probably do want inheritance eventually**: I want to be able to do this:

```jsonc
"profiles": [
	{
		"guid": "{defg}",
		"name": "Ubuntu",
		"source": "Microsoft.Terminal.Wsl" // AUTOGENERATED DYNAMIC PROFILE
	},
	{
		"name": "Ubuntu (dark theme)",
		"inheritsFrom": "{defg}", // Ubuntu
		"background": "#000000",
		"colorScheme": "Really quite dark",

		// I don't know the commandline: it's based on the ubuntu profile.
		// I shouldn't _need_ to know it.

		"source": "Microsoft.Terminal.Wsl"
	}
]
```

that is out of spec for this proposal.

Reply:
I concur. 4 seems like the weakest proposal on paper, _for this feature_. Good overall, but _far_ too heavy-handed for "these settings apply to every profile".

Reply:
It seems like `source` and `inheritsFrom` are kinda trying to do the same thing then. What if we combined them? `source` is either a...
- `string` (i.e.: `Microsoft.Terminal.WSL` or other dynamic profile ones)
- `guid`: directly inherit the resolved profile with this guid

We already check for duplicate guids so we know it's a one-to-one relationship between a guid and a profile.

Reply:
Argh. I want it but you're probably right about it being too heavy-handed for this feature. 😭